core/src/main/java/com/linecorp/armeria/client/DnsResolverGroupBuilder.java

```diff
@@ -361,13 +376,17 @@ RefreshingAddressResolverGroup build(EventLoopGroup eventLoopGroup) {
                     new DefaultDnsQueryLifecycleObserverFactory(
                             meterRegistry,
                             new MeterIdPrefix("armeria.client.dns.queries"));
-            if (dnsQueryLifecycleObserverFactory == null) {
-                builder.dnsQueryLifecycleObserverFactory(observerFactory);
-            } else {
-                builder.dnsQueryLifecycleObserverFactory(
-                        new BiDnsQueryLifecycleObserverFactory(observerFactory,
-                                                               dnsQueryLifecycleObserverFactory));
+
+            if (!disableDnsQueryLifecycleObserverFactory) {
```

Comment:
I guess there might be a chance that a user wants to set a customized `DnsQueryLifecycleObserver` while he/she doesn't want to use the `DefaultDnsQueryLifecycleObserverFactory`.
So how about changing the parameter to `disableDnsQueryMetric` and allow users to use the customized factory?

Reply:
+1 Nice idea.